File: Rearrange a string - GFG/rearrange-a-string.java
// { Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GfG
{
    public static void main(String args[])
        {
            Scanner sc = new Scanner(System.in);
            int t = sc.nextInt();
            while(t-->0)
                {
                    String s = sc.next();
                    Solution obj = new Solution();
                    System.out.println(obj.arrangeString(s));
                }
                
        }
}
// } Driver Code Ends


//User function Template for Java

class Solution
{
    public String arrangeString(String s)
        {
            //code here.
            int sum = 0  ; 
            for(int i=0;i<s.length();i++){
                
                if(s.charAt(i)>='0' && s.charAt(i)<='9'){
                    sum+=s.charAt(i)-'0'  ; 
                }
            }
            
            char ch[] = s.toCharArray()  ;  
            Arrays.sort(ch)  ; 
            String res=""  ; 
            for(int i=0;i<ch.length;i++){
                if(ch[i]>='A' && ch[i]<='Z'){
                    res+=ch[i]  ; 
                }
            }
            
            String j = sum + "" ;  
            return (res+j) ; 
        }
}

File: Rearrange a string - GFG/rearrange-a-string.cpp
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends


class Solution
{
  public:
    string arrangeString(string str)
    {
        //code here.
        
        int n = str.size()  ;
        int sum = 0  ;
        vector<char>rr ; 
        
        for(int i=0;i<n;++i){
            if(str[i]>='0' && str[i]<='9'){
                sum+=(str[i]-'0')  ; 
            }
            else{
                rr.push_back(str[i]) ;  
            }    
        }
        
        sort(rr.begin(),rr.end())  ; 
        string res=""  ; 
        for(int i=0;i<rr.size();i++){
            res+=rr[i] ;  
        }
        
        string kk = to_string(sum)  ; 
        res+=kk ; 
        
        return res; 
        
    }
};

// { Driver Code Starts.

int main(){
    int t;
    cin >> t;
    while(t--){
        string s;
        cin >> s;
        Solution ob;
        cout <<ob.arrangeString(s) << endl;
    }
return 0;
}

  // } Driver Code Ends